src/CSI.IBTA.UserService/Services/EmployersService.cs

```diff
@@ -172,13 +172,34 @@ public async Task<GenericResponse<IEnumerable<EmployerDto>>> GetAll()
             return new GenericResponse<IEnumerable<EmployerDto>>(null, res.Select(_mapper.Map<EmployerDto>));
         }
 
+        public async Task<GenericResponse<PagedEmployersResponse>> GetEmployersFiltered(int page = 1, int pageSize = 8, string nameFilter = "", string codeFilter = "")
+        {
+            var filteredEmployers = _unitOfWork.Employers.GetSet()
+            .Where(e => (nameFilter == "" || e.Name.Contains(nameFilter))
+                && (codeFilter == "" || e.Code.Equals(codeFilter)));
+
+            var totalCount = filteredEmployers.Count();
+            var totalPages = (int)Math.Ceiling((double)totalCount / pageSize);
+
+            var employers = await filteredEmployers
+                .OrderBy(p => p.Id)
+                .Skip((page - 1) * pageSize)
+                .Take(pageSize)
+                .ToListAsync();
+
+            var responseList = new PagedEmployersResponse(employers.Select(_mapper.Map<EmployerDto>).ToList(), page, pageSize, totalPages, totalCount);
+
+            if (employers == null) return new GenericResponse<PagedEmployersResponse>(new HttpError("Server failed to fetch employers", HttpStatusCode.InternalServerError), null);
```

Comment:
this is not executed as line 190 does select from "employers".
The recommendation would be to drop this line at all. If server fails to fetch it throws time out exception or any other more meaningful error.